### DISCUSSION TITLE
Create a new device share using a backup share on self-host (tKey)

### DISCUSSION THREAD

**Discussion by cesariomartins** [Web3Auth/Web3Auth#1241] [Help]
My user is backing up the share C using serialized module from tKey module.
When the user doesn't have share B (device share), how do I input the back up share so the user can create another device share?

In the code below, the backupSeed is the backup seed prase. We get the backup share using tkeyInstance.modules.shareSerialization.deserialize(backupSeed, "mnemonic");
After the tkeyInstance.inputShare(deserializedShare) call, the number of required share is still "1", so the backup share is not being viewed as a valid share. 
```
const deserializedShare = await tkeyInstance.modules.shareSerialization.deserialize(backupSeed, "mnemonic");
await tkeyInstance.inputShare(deserializedShare);
```

What is the correct way to input the share from backup?

**Comment by YZhenY** [comment]
what is the format of your backup share? are you using the correct serialization here?

**Reply by giovanna-afarelli** [reply]
This is the code for serializing the share:
`const serializedShare = await tkey.modules.shareSerialization.serialize(shareStore.newShareStores[1].share.share, "mnemonic");`
This returns a String with the backup phrase we send to the user.

This is the code for deserialization:
`const deserializedShare = await tkey.modules.shareSerialization.deserialize(seed, "mnemonic");`

After deserializing, we try to input the generated deserializedShare into the tkey instance:
`await tkey.inputShare(deserializedShare);`

There is no exception, the code seems to work, but the share does not appear in the _tkey.getKeyDetails()_ (it still shows _requiredShares = 1_)

**Reply by YZhenY** [reply]
use inputShareSafe, with tkey.inputShare doesn't catch up the share to the latest polynomial in the event you've updated it/changed its configuration.
